R/getTimeDivergence.R

```diff
@@ -0,0 +1,105 @@
+#' Beta diversity calculation within individuals in the interval of n time steps
+#'
+#' The dissimilarity (beta diversity) is calculated in time steps of n, between
+#' the samples of subject(individuals).Time interval can be 1 or greater than 1.
+#' The time difference between n points is also calculated. Given input,
+#' `SummarizedExperiment` or `TreeSummarizedExperiment` object, returns with
+#' the calculations inserted in columns of `colData` field.
+#'
+#' @param se A
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' object or
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' @param field a single character value for specifying which grouping
+#' factor chosen from columns of `colData` field.
+#' @param time_field a single character value for specifying time series
+#' vector given in `colData` field.
+#' @param time_interval integer value indicating the increment between the time
+#' steps. It can be 1 or higher than 1. (default: \code{time_interval = 1})
+#'
+#' @return a
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' or
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' containing the beta diversity and time difference between samples
+#' in n time steps
+#'
+#' @importFrom SEtools mergeSEs
+#' @importFrom SummarizedExperiment assay
+#' @importFrom SummarizedExperiment colData
+#' @importFrom SummarizedExperiment colData<-
+#'
+#' @examples
+#' library(miaTime)
+#' data(hitchip1006)
+#'
+#' hitchipTime <- getTimeDivergence(hitchip1006, field = "subject",
+#'                                    time_interval = 1,
+#'                                    time_field = "time")
+#'
+#'@export
+getTimeDivergence <- function(se, field, time_field, time_interval){
+
+# Define the names for the newly added fields
+# for the time difference and community dissimilarity
+new_field <- "time_divergence"
+new_field2 <- "time_difference"
+
+# Dissimilarity will be calculated between rows, as in stats::dist
+check_pairwise_dist <- function (x,
+                                 distfun,
+                                 time_interval,
+                                 new_field,
+                                 new_field2,
+                                 time_field){
+
+    mat <- t(assay(x))
```

Comment:
assay(x) does not specify which abundance values ie. counts, relabundance, clr.. other..? should be passed as function argument. Also check if there are other similar cases in the code.